test/unit/orchestration/dns_test.rb

```diff
@@ -82,4 +82,41 @@ def test_bmc_should_have_valid_dns_records
       assert_equal false, h.reverse_dns?
     end
   end
+
+  def test_host_should_rebuild_dns
+    h = FactoryGirl.create(:host, :with_dns_orchestration)
+    Nic::Managed.any_instance.expects(:del_dns_a_record_safe)
+    Nic::Managed.any_instance.expects(:del_dns_ptr_record_safe)
+    Nic::Managed.any_instance.expects(:recreate_a_record).returns(true)
```

Comment:
I think these should all be stubbed one level lower, so the recreate_* methods get some test coverage.  Stub `:set_dns_a_record` and `:del_dns_a_record` instead.